app/controllers/UserDataController.scala

```diff
@@ -28,35 +25,30 @@ class UserDataController(frontsApi: FrontsApi, dynamoClient: DynamoDbClient, val
     x => (Json.stringify(x))
   )
   implicit val trail: DynamoFormat[Trail] = deriveDynamoFormat[Trail]
+  implicit val cardType: DynamoFormat[CardType] = deriveDynamoFormat[CardType]
+  implicit val editionsCard: DynamoFormat[EditionsClientCard] = deriveDynamoFormat[EditionsClientCard]
   implicit val userData: DynamoFormat[UserData] = deriveDynamoFormat[UserData]
   private lazy val userDataTable = Table[UserData](config.faciatool.userDataTable)
 
-  private def updateClipboardContentByFieldName(articles: Option[JsValue], userEmail: String, fieldName: String) = {
-    val clipboardArticles: Option[List[Trail]] = articles.flatMap(jsValue =>
-      jsValue.asOpt[List[Trail]])
-
-    clipboardArticles match {
-      case Some(articles) => {
-
-        Scanamo(dynamoClient).exec(userDataTable.update(UniqueKey("email" === userEmail), set(fieldName, articles)))
+  private def updateClipboardContentByFieldName[T](maybeJson: Option[JsValue], userEmail: String, fieldName: String)(implicit dynamoFormat: DynamoFormat[T], jsonFormat: Reads[T]) = {
+    val a = maybeJson.map(_.validate[T])
```

Comment:
This `val a` looks suspiciously like a leftover debugging aid. Because I do it all the time :)

Reply:
Yep, getting a line location so I can have the debugger produce a result (macros mean evaluating in-band does not work) – unsure if there's another way to do this – thanks for spotting!
```suggestion
```

Reply:
Removed in [6eb392c](https://github.com/guardian/facia-tool/pull/1584/commits/6eb392c490ab62aaffc0c3609e0792162212dbe0).